pkg/clusteragent/admission/mutate/autoinstrumentation/config.go

```diff
@@ -60,6 +139,16 @@ func NewInstrumentationConfig(datadogConfig config.Component) (*InstrumentationC
 		return nil, fmt.Errorf("apm.instrumentation.enabled_namespaces and apm.instrumentation.disabled_namespaces are mutually exclusive and cannot be set together")
 	}
 
+	// Ensure both enabled namespaces and targets are not set together.
+	if len(cfg.EnabledNamespaces) > 0 && len(cfg.Targets) > 0 {
+		return nil, fmt.Errorf("apm.instrumentation.enabled_namespaces and apm.instrumentation.targets are mutually exclusive and cannot be set together")
+	}
+
+	// Ensure both library versions and targets are not set together.
+	if len(cfg.LibVersions) > 0 && len(cfg.Targets) > 0 {
+		return nil, fmt.Errorf("apm.instrumentation.lib_versions and apm.instrumentation.targets are mutually exclusive and cannot be set together")
```

Comment:
🤔 What are your thoughts on these being the default versions we'd use for a target if none were specified at the `Target` level?

Reply:
I suspect it will be confusing to the user and be less explicit, especially given `lib_versions` is used today to "target" workloads.

Reply:
After thinking about this more, I think config values should NOT change meaning. For example, in the targets based selection, we have decided to keep `disabled_namespaces` and that's ok given the meaning doesn't change. In the case of using lib_versions to determine the default versions, it would be a behavior change from what it means today.